tidy3d/components/simulation.py

```diff
@@ -2147,7 +2147,12 @@ def dt(self) -> float:
         dl_mins = [np.min(sizes) for sizes in self.grid.sizes.to_list]
         dl_sum_inv_sq = sum((1 / dl**2 for dl in dl_mins))
         dl_avg = 1 / np.sqrt(dl_sum_inv_sq)
-        return self.courant * dl_avg / C_0
+        # material factor
+        n_cfl = 1.0
+        for mat in self.mediums:
+            n_cfl = min(n_cfl, mat.n_cfl)
```

Comment:
a bit pedantic but probably better and more pythonic is `n_cfl = min([mat.n_cfl for mat in self.mediums], default = 1.0)`

Reply:
More importantly, shouldn't we actually be able to remove the `default = 1.0`? `self.mediums` is never empty as it contains at least the simulation medium, and if all media in the simulation have n_cfl > 1, we could actually increase the time step?

Reply:
Tried a few simulations with dispersiveless medium, and all converges. But I think it might be problematic for dispersive medium for how we decide to apply supixel or not based on permittivity > 1.

Reply:
Oh, very good point. Ok yeah let's leave 1 as the minimum.